cardano-cli/src/Cardano/CLI/Type/Error/TxValidationError.hs

```diff
@@ -62,81 +60,48 @@ validateScriptSupportedInEra era script@(ScriptInAnyLang lang _) =
           (anyCardanoEra $ toCardanoEra era)
     Just script' -> pure script'
 
-data TxNotSupportedInEraValidationError era
-  = -- | First argument is the kind of data that is not supported.
-    -- Second argument is the era that doesn't support the data.
-    TxNotSupportedInAnyCardanoEraValidationError T.Text AnyCardanoEra
-
-instance Show (TxNotSupportedInEraValidationError era) where
-  show =
-    \case
-      TxNotSupportedInAnyCardanoEraValidationError a cEra -> go a cEra
-   where
-    go a era = show (pretty a) <> " not supported in " <> show era
-
-instance Error (TxNotSupportedInEraValidationError era) where
-  prettyError =
-    \case
-      TxNotSupportedInAnyCardanoEraValidationError a cEra -> go a cEra
-   where
-    go a cEra = pretty a <+> "not supported in" <+> viaShow cEra
-
 validateTxTotalCollateral
-  :: ShelleyBasedEra era
-  -> Maybe Lovelace
-  -> Either (TxNotSupportedInEraValidationError era) (TxTotalCollateral era)
-validateTxTotalCollateral _ Nothing = return TxTotalCollateralNone
-validateTxTotalCollateral sbe (Just coll) = do
-  supported <-
-    conjureWitness (toCardanoEra sbe) $
-      TxNotSupportedInAnyCardanoEraValidationError "Transaction collateral"
-  pure $ TxTotalCollateral supported coll
+  :: IsEra era
+  => Maybe Lovelace
+  -> TxTotalCollateral era
+validateTxTotalCollateral Nothing = TxTotalCollateralNone
+validateTxTotalCollateral (Just coll) = do
+  TxTotalCollateral (convert useEra) coll
```

Comment:
I think we should remove those trivial conversion functions which are called `validate*`

Reply:
100%. I will do that in a follow up PR.